src/app.js

```diff
@@ -25,6 +25,10 @@ var OffsetToLineColumnConverter = require('./lib/offsetToLineColumnConverter')
 
 var examples = require('./app/example-contracts')
 
+// components
+var fileExplorer = require('_file-explorer')
```

Comment:
1. you can require("_modulename") from everywhere within the project's `./src` folder.

2. From my experience stuff like `app-module-path` breaks in the browser, because browserify doesn't know how to deal with this. In the past I wrote `https://www.npmjs.com/package/rootrelative` but that was a naive approach which of course - again - doesnt work in browsers :/
3. you do not need an actual folder and a package.json for simple modules. The simplest form would actually be just `./src/node_modules/_my-private-module.js` ... a next step (for more complex private modules would be `./src/node_modules/_my-private-module/index.js` which results in lots of open editor tabs with the name "index.js" (which sucks) ...a step up from that would be `./src/node_modules/_my-private-module/my-private-module.js` + a package.json to describe `"maine": "my-private-module.js"`.

The advantage is, that a project wide simple search & replace works, because the requirepath is a constant instead of a relative path, which makes it trivial to publish an internal module to npm if it turns out a cool module should be re-used across different projects :-)